xbmc/interfaces/python/PythonInvoker.cpp

```diff
@@ -641,15 +637,14 @@ void CPythonInvoker::onPythonModuleInitialization(void* moduleDict)
 
   PyObject* moduleDictionary = (PyObject*)moduleDict;
 
-  PyObject* pyaddonid = PyUnicode_FromString(m_addon->ID().c_str());
-  PyDict_SetItemString(moduleDictionary, "__xbmcaddonid__", pyaddonid);
+  PyDict_SetItemString(moduleDictionary, "__xbmcaddonid__",
+                       PyObjectPtr(PyUnicode_FromString(m_addon->ID().c_str())).get());
```

Comment:
I feel like this could still be two lines, not a big deal though.
```
auto id = PyObjectPtr(PyUnicode_FromString(m_addon->ID().c_str()));
PyDict_SetItemString(moduleDictionary, "__xbmcaddonid__", id.get());
```

Reply:
This C++ syntax is not optimal, but it does the job.